tests/test_hierarchy.cc

```diff
@@ -193,35 +195,51 @@ BOOST_DATA_TEST_CASE(
 
     // This is a gold standard test. Not the greatest but it makes sure we don't
     // break the code
-    std::map<std::tuple<std::string, bool, std::string>, double> ref_solution;
-    ref_solution[std::make_tuple("hyper_cube", false, "None")] =
-        is_matrix_free ? 0.1482630509 : 0.0491724046;
-    ref_solution[std::make_tuple("hyper_cube", false,
-                                 "Reverse Cuthill_McKee")] =
-        is_matrix_free ? 0.1482630509 : 0.0491724046;
-    ref_solution[std::make_tuple("hyper_cube", true, "None")] =
-        is_matrix_free ? 0.1575262224 : 0.0488984875;
-    ref_solution[std::make_tuple("hyper_cube", true, "Reverse Cuthill_McKee")] =
-        is_matrix_free ? 0.1575262224 : 0.0488984875;
-    ref_solution[std::make_tuple("hyper_ball", false, "None")] =
-        ref_solution[std::make_tuple("hyper_ball", false,
-                                     "Reverse Cuthill_McKee")] =
-            is_matrix_free ? 0.3022004744 : 0.1146629782;
-    ref_solution[std::make_tuple("hyper_ball", true, "None")] =
-        is_matrix_free ? 0.2977841482 : 0.1024334788;
-    ref_solution[std::make_tuple("hyper_ball", true, "Reverse Cuthill_McKee")] =
-        is_matrix_free ? 0.2977841482 : 0.1024334788;
-
-    if (mesh == std::string("hyper_cube"))
-      BOOST_TEST(
-          conv_rate ==
-              ref_solution[std::make_tuple(mesh, distort_random, reordering)],
-          tt::tolerance(1e-6));
-    else
-      BOOST_TEST(
-          conv_rate ==
-              ref_solution[std::make_tuple(mesh, distort_random, reordering)],
-          tt::tolerance(1e-6));
+    //   (geometry, distortion, reordering, eigensolver, matrix-free)
+    std::map<std::tuple<std::string, bool, std::string, std::string, bool>,
+             double>
```

Comment:
Readability: I would prefer a string or a scoped enumeration to denote matrix-free mode.